### PR TITLE
tools/kvstore-tool: implement 'histogram' command

### PR DESCRIPTION
**Pull request by ifed01**
This command shows information on key/value counts and lengths for KV  DB.

Signed-off-by: Igor Fedotov <ifedotov@suse.com>

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/tools/ceph_kvstore_tool.cc [hunk 1]
```diff
@@ -347,6 +348,8 @@ int main(int argc, const char *argv[])
     st.compact_range(prefix, start, end);
   } else if (cmd == "stats") {
     st.print_stats();
+  } else if (cmd == "dissect") {
+    st.dissect();
```
Comment (aclamk) [inline src/tools/ceph_kvstore_tool.cc:352]: I feel like "dissect" sounds unintuitive. 
How about "kv_hist[ogram]". But is is just a proposal.

File: src/tools/kvstore_tool.cc [hunk 2]
```diff
@@ -228,6 +230,66 @@ int StoreTool::print_stats() const
...
+
+  size_t max_key_size = 0, max_value_size = 0;
+  uint64_t total_key_size = 0, total_value_size = 0;
+  size_t key_size = 0, value_size = 0;
+  KeyValueHistogram hist;
+
+  auto start = coarse_mono_clock::now();
+
+  KeyValueDB::WholeSpaceIterator iter = db->get_wholespace_iterator();
+  iter->seek_to_first();
```
Comment (aclamk) [inline src/tools/kvstore_tool.cc:250]: Is it difficult to be able to provide size histogram stats for selected prefix?

**Review by aclamk — CHANGES_REQUESTED** [review]
- [ ] update documentation

I think it could be improved by adding ability for histogram for single prefix.

**Comment by ifed01** [comment]
@aclamk - resolved your comments, please take a look

**Review by aclamk — APPROVED** [review]
Great stuff.

**Pull request merged**